lib/connection.js

```diff
@@ -195,6 +216,9 @@ export default function Connection(options = {}) {
     clearTimeout(connect_timer)
     if (err) {
       if (backend.query) {
+        if (!backend.query.retried && retryRoutines[err.routine])
+          return await retry(backend.query)
```

Comment:
Useless `await` (not really important but can help for maintenance)
```suggestion
          return retry(backend.query)
```